.github/workflows/helm-chart-ci.yaml

```diff
@@ -165,39 +144,79 @@ jobs:
         run: |
-          post-install() {
-            [ -x "${TEST_DIR}/post-install.sh" ] && "${TEST_DIR}/post-install.sh" $1
-            exit $1
-          }
-
-          trap 'post-install $? $LINENO' EXIT
-
-          export scenario="$(basename "${TEST_DIR}")"
-          export EXTRA_HELM_ARGS=""
-
-          source .github/tests/charts.sh
-
-          [ "${scenario}" != "default" ] && kubectl create namespace "${scenario}"
-          [ -x "${TEST_DIR}/pre-install.sh" ] && "${TEST_DIR}/pre-install.sh"
-          [ -f "${TEST_DIR}/.env" ] && source "${TEST_DIR}/.env"
-
-          if [ -x "${TEST_DIR}/install.sh" ]; then
-            "${TEST_DIR}/install.sh"
-          else
-            ct install --debug \
-              --charts "charts/spire" \
-              --namespace "${scenario}" \
-              --target-branch ${{ github.base_ref }} \
-              --exclude-deprecated \
-              --skip-clean-up \
-              ${{ (matrix.values != 'default' && '--helm-extra-set-args "--values=${TEST_DIR}/values.yaml ${EXTRA_HELM_ARGS}"') || '' }}
-          fi
-        env:
-          TEST_DIR: .github/tests/${{ matrix.values }}
+          ct install --config ct.yaml \
+            --target-branch ${{ github.base_ref }}
+
+      - name: Test summary
+        if: always()
+        run: ./post-install.sh
+        working-directory: .github/tests
+
+  build-matrix:
+    name: Build matrix
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.5.2
+
+      - id: set-matrix
+        name: Collect all examples
+        run: |
+          examples="$(find examples -maxdepth 2 -type f -name run-tests.sh | xargs -I % dirname %)"
+          examples_json="$(echo "$examples" | jq -c --slurp --raw-input 'split("\n") | map(select(. != ""))')"
+          echo "${examples_json}"
+          echo "examples=$examples_json" >>"$GITHUB_OUTPUT"
+
+    outputs:
+      examples: ${{ steps.set-matrix.outputs.examples }}
+
+  example-test:
+    runs-on: ubuntu-22.04
+
+    needs:
+      - lint-chart
+      - build-matrix
+
+    strategy:
+      fail-fast: false
+      matrix:
+        example:
+          - ${{ fromJson(needs.build-matrix.outputs.examples) }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.5.2
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3.5
+        with:
+          version: ${{ env.HELM_VERSION }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.7.0
+        # Only build a kind cluster if there are chart changes to test.
+        with:
+          version: ${{ env.KIND_VERSION }}
+          node_image: kindest/node:v1.26.4
+          config: .github/kind/conf/kind-config.yaml
+          verbosity: 1
+
+      - name: Install and test example
+        run: ${{ matrix.example }}/run-tests.sh
```

Comment:
Looking at the pr action run list, I don't see this run per the versions we support. Something missing iterator thing here?

Reply:
Intentionally only test the examples on the latest version, all the ci/*-values.yaml are tested on all versions.

Examples are essentially a combination of what is already tested there, so therefore saving some compute here on these tests.

Reply:
I would think the examples are the ones we most want to ensure work on any cluster the user might be deploying on, so should be well tested on various versions too.

Reply:
The examples are just a combination of the ci tests. Those already cover the same ground and would already cause the failure in case the other k8s version wouldn't support things. It would just be a waste of resources to run the examples again on 3 k8s versions.

Reply:
I don't think thats true... take for example the production example. If we make a change to it, and we're only testing on the newest, it may break for n-1 and we wouldn't know it. I don't see any of the ct test cases with the same functionality, so its not just a combination of existing ct tested features.

Reply:
Which exact ones do you mis?